daemon/build.go

```diff
@@ -0,0 +1,125 @@
+package daemon
+
+import (
+	"github.com/Sirupsen/logrus"
+	"github.com/docker/distribution/reference"
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/backend"
+	"github.com/docker/docker/builder"
+	"github.com/docker/docker/image"
+	"github.com/docker/docker/layer"
+	"github.com/docker/docker/pkg/stringid"
+	"github.com/docker/docker/registry"
+	"github.com/pkg/errors"
+	"golang.org/x/net/context"
+	"io"
+)
+
+type releaseableLayer struct {
+	rwLayer layer.RWLayer
+	release func(layer.RWLayer) error
+	mount   func() (layer.RWLayer, error)
+}
+
+func (rl *releaseableLayer) Release() error {
+	if rl.rwLayer == nil {
+		return nil
```

Comment:
`roLayer` still needs to be released. Might be simpler to add a ref to `roLayer` in `releaseableLayer` instead of defining these callback methods.